File: 704 - Binary Search.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        i = 0
        j = len(nums) - 1
        res = -1
        while i <= j:
            mid = (i + j) // 2
            # print(mid)
            if target > nums[mid]:
                i = mid + 1
            elif target < nums[mid]:
                j = mid - 1
            else:
                return mid
            # print(i, j)
        return res


s = Solution()
a = s.search([-1, 0, 3, 5, 9, 12], 9)
print(a)

File: 704 - Binary Search.java
class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length - 1, mid, res = -1; 
        while (l <= r) {
            mid = (l + r) / 2;
            if (nums[mid] < target) {
                l = mid + 1;
            } else if (nums[mid] > target) {
                r = mid - 1;
            } else {
                return mid;
            }
        }
        return res;
    }
}